### PR TITLE
feat(middlewared/schema): List should not have a default

### PR DESCRIPTION
**Pull request by william-gr**
Having a default is confusing and may break usage when using within a
Dict, making it so we cant know if a value has been provided by the user
or not.

Ticket:	#35710

### CODE REVIEW

File: src/middlewared/middlewared/plugins/account.py [hunk 1]
```diff
@@ -135,7 +135,7 @@ class Config:
         if verrors:
             raise verrors
 
-        groups = data.pop('groups') or []
+        groups = data.pop('groups', None) or []
```
Comment (themylogin) [inline src/middlewared/middlewared/plugins/account.py]: Why no `default=[]` here?

**Comment by william-gr** [comment]
@themylogin changed to default=[] in List and removed .pop line

**Comment by themylogin** [comment]
@william-gr great job, 11.3 will be solid!

**Comment by william-gr** [comment]
Thanks for the thorough review, guys!

**Pull request merged**